### PR TITLE
Add rolebinding for the argocd-manager SA to deploy to the namespace and add apiGroups

### PR DESCRIPTION
**Pull request by maulikjs**

### CODE REVIEW

File: data-catalog/base/argocd-rb.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: authorization.openshift.io/v1
+kind: RoleBinding
+metadata:
+  name: argocd-manager-rolebinding
+  namespace: dev
```
Comment (accorvin) [inline data-catalog/base/argocd-rb.yaml]: I believe that the namespace should not be specified here

File: data-catalog/overlays/dev/argocd-rb.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+apiVersion: authorization.openshift.io/v1
+kind: RoleBinding
+metadata:
+  name: argocd-manager-rolebinding
+  namespace: dh-dev-analytics-factory
```
Comment (accorvin) [inline data-catalog/overlays/dev/argocd-rb.yaml]: I think the namespace should be specified in the kustomize file for all of these, not on the objects.

**Comment by HumairAK** [comment]
none of the rolebindings are included in the kustomizations (i.e. they don't get outputed via kustomize build) -- is that intentional?

File: data-catalog/overlays/dev/argocd-rb.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+apiVersion: authorization.openshift.io/v1
+kind: RoleBinding
```
Comment (accorvin) [inline data-catalog/overlays/dev/argocd-rb.yaml]: actually, you shouldn't need this at all, right? Since you already have it in the base above

Comment (HumairAK) [reply data-catalog/overlays/dev/argocd-rb.yaml]: I think he should remove it from the base

**Comment by HumairAK** [comment]
this may be out of scope of this pr, but none of the root kustomizations seem to specify the namespace, something worth considering adding here or in a separate pr

**Review by accorvin — APPROVED** [review]

**Pull request merged**